ebpf/tracer.go

```diff
@@ -136,16 +216,42 @@ func (t *Tracer) getConnections() ([]ConnectionStats, error) {
 	// Remove expired entries
 	t.removeEntries(mp, tcpMp, expired)
 
-	return active, nil
+	// Add the closed connections
+	t.closedConnsLock.Lock()
+	for i := 0; i < len(t.closedConns); i++ {
+		t.closedConns[i].collected = true
+	}
+	active = append(active, t.closedConns...)
+	t.closedConnsLock.Unlock()
+
+	return removeDuplicates(active), nil
+}
+
+func (t *Tracer) cleanupClosedConns() {
+	newClosedConns := []ConnectionStats{}
+	t.closedConnsLock.Lock()
+	for _, c := range t.closedConns {
+		if !c.collected {
+			newClosedConns = append(newClosedConns, c)
+		}
+	}
+	t.closedConns = newClosedConns
+	t.closedConnsLock.Unlock()
 }
 
 func (t *Tracer) removeEntries(mp, tcpMp *bpflib.Map, entries []*ConnTuple) {
-	for i := range entries {
-		t.m.DeleteElement(mp, unsafe.Pointer(entries[i]))
+	var e *ConnTuple
+	for _, ent := range entries {
+		e = ent
```

Comment:
I don't quite understand the need for this. Isn't `ent` a pointer already?

Reply:
Did you mean for this to be `entries[i]`?

Reply:
woops right !